eng/pipelines/templates/stages/platform-matrix.json

```diff
@@ -59,13 +59,23 @@
       "Agent": {
         "ubuntu-20.04": { "OSVmImage": "MMSUbuntu20.04", "Pool": "azsdk-pool-mms-ubuntu-2004-general" }
       },
-      "JavaTestVersion": "1.20",
+      "JavaTestVersion": "1.17",
+      "AZURE_TEST_HTTP_CLIENTS": "netty",
+      "TestFromSource": false,
+      "RunAggregateReports": false,
+      "TestGoals": "surefire:test",
+      "TestOptions": ""
+    },
+    {
+      "Agent": {
+        "ubuntu-20.04": { "OSVmImage": "MMSUbuntu20.04", "Pool": "azsdk-pool-mms-ubuntu-2004-general" }
+      },
+      "JavaTestVersion": "1.21",
       "AZURE_TEST_HTTP_CLIENTS": "netty",
       "TestFromSource": false,
       "RunAggregateReports": false,
       "TestGoals": "surefire:test",
-      "TestOptions": "",
-      "IsLatestNonLtsJdk": true
+      "TestOptions": ""
```

Comment:
Oh right, let's remove the duplicate run. That said, I'm a little unsure about if we even need to run tests using the latest non-LTS version of Java. I get why we did it when Java 20 was the latest version of the language, but do we even need to do that now?

Reply:
The idea was, while the latest JDK available isn't LTS add in a custom run for the latest JDK to have speculative checks for things that will break once the next LTS is released. For example, if that was added with more time before Java 21 was released, it would have given us time to fix all Javadocs and once Java 21 was released we'd just flip a switch over to it, rather than what we need to do right now which is fixing everything before we can flip that switch.

And once Java 22 is released we can begin the process again. The reason it is something custom is the CI agents we use only provide LTS JDKs, everything else is something we have to install on the fly.